File: backend/src/main/java/com/freelancer/service/UserService.java
package com.freelancer.service;

import com.freelancer.dto.request.PaymentInfoRequest;
import com.freelancer.dto.request.UserProfileUpdateRequest;
import com.freelancer.dto.response.PaymentInfoResponse;
import com.freelancer.dto.response.UserProfileResponse;

import java.util.List;

public interface UserService {

    UserProfileResponse getProfile(Long userId);

    UserProfileResponse updateMyProfile(Long userId, UserProfileUpdateRequest req);

    List<PaymentInfoResponse> getMyPaymentInfos(Long userId);

    PaymentInfoResponse savePaymentInfo(Long userId, PaymentInfoRequest req);

    void deletePaymentInfo(Long userId, Long paymentInfoId);
}

File: frontend/src/app/core/services/user.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { ApiService } from './api.service';
import { UserProfile, UserProfileUpdateRequest, PaymentInfo, PaymentInfoRequest } from '../models/user.model';

@Injectable({ providedIn: 'root' })
export class UserService {
  private api = inject(ApiService);

  getProfile(id: number | string): Observable<UserProfile> {
    return this.api.get<UserProfile>(`/users/${id}`);
  }

  updateMyProfile(req: UserProfileUpdateRequest): Observable<UserProfile> {
    return this.api.put<UserProfile>('/users/me/profile', req);
  }

  getMyPaymentInfos(): Observable<PaymentInfo[]> {
    return this.api.get<PaymentInfo[]>('/users/me/payment-info');
  }

  savePaymentInfo(req: PaymentInfoRequest): Observable<PaymentInfo> {
    return this.api.post<PaymentInfo>('/users/me/payment-info', req);
  }

  deletePaymentInfo(id: number): Observable<void> {
    return this.api.delete<void>(`/users/me/payment-info/${id}`);
  }
}
